### PR TITLE
Create initial version of js object to JSON

### PR DESCRIPTION
**Pull request by tlbdk**

### CODE REVIEW

**Review by mex — APPROVED** [review]

File: src/json.ts [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+  | RegExp
+  | { new (...args: JavaScriptValue[]): JavaScriptValue }
+  | ((...args: JavaScriptValue[]) => JavaScriptValue)
+
+export function objectToJson(
+  jsValue: JavaScriptValue,
+  options: Partial<ObjectToJsonOptions> & { maxDepth?: number } = {}
+): Json {
+  const seen: JavaScriptValue[] = []
+  const maxStringLength =
```
Comment (Viterkim) [inline src/json.ts:25]: maybe for readability make the function bigger? (to make it easier to read)

File: src/json.ts [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+      if (!isFinite(jsValue)) {
+        return '(Infinity)'
+      }
+      return jsValue
+    }
+    case 'bigint':
+      return `BigInt(${jsValue?.toString()})`
+    case 'string':
+      return (
+        jsValue.replace(/\n/g, '\\n').substring(0, options.maxStringLength) +
```
Comment (Viterkim) [inline src/json.ts:85]: maybe a comment

File: src/journald.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  if (getLogLevel() <= level) {
+    const output = args
+      .map(a => {
+        const o = objectToJson(a as JavaScriptValue)
+        return typeof o === 'string' ? o : JSON.stringify(o)
+      })
+      .join(' ')
+    const journaldLevel = logLevelToJournaldLevel(level)
+    // eslint-disable-next-line no-console
+    console.log(`<${journaldLevel}> ${output}`)
```
Comment (Viterkim) [inline src/journald.ts:50]: Maybe add a comment here to explain how this works?

**Review by Viterkim — APPROVED** [review]

File: src/journald.ts (hunk 3 above)
Comment (ofweb) [reply src/journald.ts:50]: We still need to se that it works, but https://www.loggly.com/blog/logging-in-new-style-daemons-with-systemd/

**Comment by ofweb** [comment]
We tested on a device and works great 
image

**Comment by Viterkim** [comment]
Did you also try filtering? 
journalctl -b -p 0..5 -u ccapp@*

File: src/journald.ts [hunk 4]
```diff
@@ -47,6 +47,6 @@ export function logJournald(level: number, ...args: unknown[]): void {
       .join(' ')
     const journaldLevel = logLevelToJournaldLevel(level)
     // eslint-disable-next-line no-console
-    console.log(`<${journaldLevel}> ${output}`)
+    console.log(`<${journaldLevel}> [${getLogLevelName(level)}] ${output}`)
```
[the remark this answers is not in the document]
Comment (ofweb) [reply src/journald.ts:50]: Yes, the first part in `<>` is for journald so it knows what log level the line is.
The second part is for us so it is easier to know when you save log lines to a file.

File: src/journald.ts [hunk 5]
```diff
@@ -46,6 +46,7 @@ export function logJournald(level: number, ...args: unknown[]): void {
       })
       .join(' ')
     const journaldLevel = logLevelToJournaldLevel(level)
+    // `<${journaldLevel}> is used by journald to determine log level
```
Comment (Viterkim) [inline src/journald.ts:49]: nice

File: src/json.ts [hunk 6]
```diff
@@ -36,7 +36,7 @@ function intFromEnv(envName: string): number | undefined {
     return
   }
   const parsed = parseInt(value)
-  if (parsed === -1) {
+  if (isNaN(parsed)) {
```
Comment (Viterkim) [inline src/json.ts:39]: isnt this taken care of in parseInt ? what about infinity / minus infinity (checkFinite)

**Review by Viterkim — APPROVED** [review]

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@connectedcars/logutil",
-  "version": "5.0.3",
+  "version": "6.0.0",
```
Comment (mex) [inline package.json:3]: We usually don't update the version until after the PR has been approved and merged into master (that way we can use the `npm version` command which automatically updates the package files and creates git tag).

File: src/json.ts (hunk 6 above)
Comment (ofweb) [reply src/json.ts:39]: This is the standard parseInt that will return NaN if the string is not a number.

File: package.json (hunk 7 above)
Comment (mex) [reply package.json]: No, we just push that version bump directly to master. The teams that usually need to bump the versions on each NPM package repo has permission to bypass the branch protection. I've added the device team as well to this repo and there is a step-by-step guide for publishing a new version here: https://wiki.connectedcars.io/backend/packages/npm#updating-subsequent-publishes

**Review by mex — APPROVED** [review]

**Pull request merged**